docs/projects/tad/adrs/0005-ai-verify-technical-tests.md

```diff
@@ -0,0 +1,53 @@
+# TAD-0005 Use AI Verify for technical tests
+
+## Context
+
+The [AI Verify project](https://github.com/aiverify-foundation/aiverify) is set up in a modular way, and the
+technical tests is one of the modules. The AI Verify
+team is developing a feature which makes it possible to run the technical tests using an API: a python library with
+a method to run a test and providing the required configuration; for example, which  model and dataset to use and
+some test specific configuration.
+
+The result of the test are returned in a JSON format,
+which can be processed in any way we please, like writing it to a file or System Card or store it in a database.
+
+### Pros
+
+* We have several technical tests we can use of the shelf, like SHAP, ALE, or Fairness metrics.
+* Tests are set up in a generic way using interfaces which allows others, like ourselves, to create and add
+ their own plugins.
+* Loading models, pipelines and data is done through the AI Verify toolkit, which not only streamlines this process
+ but also performs validation and support checks.
+* We benefit from new tests being added in the future.
+
+### Cons
+
+* The outcome of an AI Verify test depends on the implementation chosen by the developer. This means
+ there's less control over what data goes into the process report compared to the final result report.
```

Comment:
> less control over what data goes into the process report compared to the final result respot

This gives me the feeling that we might lose data that could be used in the model, but that is not part of the final result report.

Could you share more about this behavior of the AI Verify test suite?

Reply:
@ravimeijerrig can you ellaborate on this question?

Reply:
My understanding is that we have limited flexibility in adjusting the content of the process and final result reports, as these forms are set by AI Verify. However, based on what we've seen so far, it appears that these reports meet our requirements.

Reply:
@ravimeijerrig, do you think we should change this con so we explain it a little bit more? I think we should write it down so that we can reproduce what we meant with this con in a couple of months.

Reply:
Hi  @robbertbos, 
 @uittenbroekrobbert and I have found a new way of describing this con. Is this more clear? 
 
 "The outcome of an AI Verify test depends on the implementation chosen by the developer of the plugin. This means there is no control or flexibility over what data (metrics, logs etc.) is included in the output."